docs/enhancements/ova-upload.md

```diff
@@ -0,0 +1,169 @@
+
+#### Story 4
+
+As a Forklift user, I want to remove an appliance from the appliance catalogue.
+
+#### Story 5
+
+As a Forklift administrator, I want to disable the appliance catalogue management endpoints.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Implementation can be approached in two stages.
+
+* 1: Supporting arbitrary PVCs as storage backing for OVA providers and corresponding UX change.
+* 2: Supporting upload and delete of appliances in the provider server and corresponding UX change.
+
+#### Supporting Arbitrary PVCs
+
+The provider controller currently expects an OVA provider resource to contain
+the path to an NFS share in its `url` field. This URL is used to construct an NFS PV
+which is mounted by an OVA provider server that is created when the OVA provider resource is reconciled.
+This field will be made optional for OVA providers, and the provider controller will be updated to look for a
+`pvc` key in the `settings` field which may contain the namespaced name of a PVC that will be mounted
+by the provider server.
+
+The current provider controller implementation does not update the deployment for the OVA provider server
+when an OVA provider resource is reconciled; the deployment is only created if it does not exist and deleted
+when an OVA provider resource is removed. The provider controller needs to be adjusted to keep OVA provider server
+deployments in sync with the storage configuration of the provider resource.
+
+This will require corresponding UX changes to relax the requirement on the URL field and expose an interface for
+setting the desired PVC.
+
+#### Uploading OVAs
+
+The OVA provider server is currently very simple, consisting of a single file using the built-in HTTP server to expose
+endpoints for consumption by the Forklift inventory. The OVA provider server needs to be extended with endpoints for
+uploading, listing, and removing appliances. 
+
+* `POST /appliances` Accepts a single OVA file to be stored in the appliance catalogue, returning a 409 if an appliance with that filename already exists.
+* `GET /appliances` Return JSON array containing metadata about the appliances in the catalog (size, upload date, etc)
+* `DELETE /appliances/:appliance` Remove an appliance from the catalogue, using OVA filename as the parameter.
+
+Due to this increase in complexity, the provider server should be refactored to use a more feature rich server
+(preferably Gin which is used in Forklift for the inventory) and reorganized into multiple files. Special consideration
+needs to be given to the upload endpoint to ensure that uploaded appliances are streamed to disk efficiently. Configuration
+options should be provided to set the maximum accepted file size.
+
+### Security, Risks, and Mitigations
+
+The upload component of this enhancement permits uploading and deleting arbitrary files from a volume in the cluster,
```

Comment:
Would we upload to one volume or each OVA have its own volume?
Might be problematic with the size requiremets

Reply:
One volume. The end user will need to ensure the volume they allocate has space enough for the OVAs they want to store. (This is already the case with the NFS share, which Forklift creates a PV from).

Reply:
hmm yeah but in that case the user has under control and knows how large the PV should be as he is the one setting it up. But in this case they could easily try to upload one additional image and could fail. I'm thinking if it would not be more user friendly that the user would provide us with storage class and we would create the PVs from that one and it would be on demand depending on the OVA size.